AK/String.h

```diff
@@ -88,6 +88,14 @@ class String {
     {
     }
 
+    template<typename F>
+    String(size_t length, F f)
+    {
+        char* buffer;
+        m_impl = StringImpl::create_uninitialized(length, buffer);
+        f(buffer);
+    }
+
```

Comment:
This does not feel like an idiomatic fit, why is this better than simply using `StringImpl::create_uninitialized()` at the call site?

Reply:
Ah I didn't realize that `StringImpl` was meant to be used outside of `String`... Thought it was internal to `String`.